pkg/trace/api/config.go

```diff
@@ -0,0 +1,8 @@
+package api
+
+// UpdateLogLevel updates HTTPReceiver unexported fields according to the
+// updated log level of trace agent
+func (r *HTTPReceiver) UpdateLogLevel(logLvl string) {
+	r.conf.LogLevel = logLvl
+	r.debug = logLvl == "debug"
```

Comment:
I'm afraid more cleanup is needed here. A few things:
1. `r.debug` field is not used anywhere as far as I can see, we can get rid of it completely
2. `r.conf.LogLevel` is also not used anywhere inside `pkg/trace`, so we should get rid of it (`cmd/trace-agent` uses it only and it does not need to be part of the `AgentConfig` structure)

As a result, we can delete this method and its call. I should've gotten rid of `LogLevel` myself but  have not noticed this aspect at the time.

Reply:
I mean, that's great then. Will do!